src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -511,6 +510,43 @@ public override void RemoveProject(ProjectId projectId)
             return getter.Diagnostics;
         }
 
+        public override bool ContainsDiagnostics(Workspace workspace, ProjectId projectId)
+        {
+            // need to improve perf in v2.
+            foreach (var stateSet in _stateManager.GetStateSets(projectId))
+            {
+                for (var stateType = 0; stateType < s_stateTypeCount; stateType++)
+                {
+                    var state = stateSet.GetState((StateType)stateType);
+                    if (state.Count == 0)
+                    {
+                        continue;
+                    }
+
+                    if (state.GetDataCount(projectId) > 0)
```

Comment:
I thought about that, but for this one, what it is doing already seems expensive. so no, but in V2, will do.